File: zig/src/lo_io_probe.zig
//! I/O round-trip driver, mirroring the Rust skeleton's probe. Reads one integer
//! from stdin and echoes it, then prints the literal 42, each on its own line.
//! Driven by the `test` build step (see build.zig). Not part of the shipped
//! runtime.

const rt = @import("lo_runtime.zig");

pub fn main() void {
    rt.init.lo_runtime_init();

    const n = rt.io.lo_read_int();
    rt.io.lo_print_int(n);
    rt.io.lo_println();

    rt.io.lo_print_int(42);
    rt.io.lo_println();

    rt.init.lo_runtime_shutdown();
}

File: cpp/tests/lo_io_probe.cpp
// I/O round-trip driver, mirroring the Rust and Zig probes. Reads one integer
// from stdin and echoes it, then prints the literal 42, each on its own line.
// Driven by tests/io_roundtrip.sh. Not part of the shipped runtime.

#include "lo_runtime/lo_runtime.h"

int main() {
  lo_runtime_init();

  const std::int32_t n = lo_read_int();
  lo_print_int(n);
  lo_println();

  lo_print_int(42);
  lo_println();

  lo_runtime_shutdown();
  return 0;
}
